git-guides/git-overview.md

```diff
@@ -0,0 +1,107 @@
+
+Whether or not you've worked with version control before, there are a few things you should know before getting started with Git:
+
+- Branches are lightweight and cheap, so it's OK to have many of them
+- Git stores changes in SHA hashes, which work by compressing text files. That makes Git a very good version control system (VCS) for software programming, but not so good for binary files like images or videos.
+- Git repositories can be connected, so you can work on one locally on your own machine, and connect it to a shared repository. This way, you can push and pull changes to a repository and easily collaborate with others.
+
+## Why Use Git?
+
+Version control is very important - without it, you risk losing your work. With Git, you can make a "commit", or a save point, as often as you'd like. You can also go back to previous commits. This takes the pressure off of you while you're working. Commit often and commit early, and you'll never have that gut sinking feeling of overwriting or losing changes.
+
+There are many version control systems out there - but Git has some major advantages.
+
+- Like we mentioned above, Git uses SHA compression, which makes it very fast.
+- Git can handle merge conflicts, which mean that it's OK for multiple people to work on the same file at the same time. This opens up the world of development in a way that isn't possible with centralized version control. You have access to the entire project, and if you're working on a branch, you can do whatever you need to and know that your changes are safe.
+- Speaking of branches, Git offers a lot of flexibility and opportunity for collaboration with branches. By using branches, developers can make changes in a safe sandbox. Instead of only committing code that is 100% sure to succeed, developers can commit code that might still need help. Then, they can push that code to the remote and get fast feedback from integrated tests or peer review. Without sharing the code through branches, this would never be possible.
+- Ease of roll back - if you make a mistake, it's OK! Commits are immutable, meaning they can't be changed. (*Note: You _can_ change history, but it will create new replacement commits instead of editing the existing commits. More on that later!*) This means that if you do make a mistake, even on an important branch like master, it's _OK_. You can easily revert that change, or roll back the branch pointer to the commit where everything was fine. The benefits of this can't be overstated. Not only does it create a safer environment for the project and code, but it fosters a development environment where developers can be braver, trusting that Git has their back.
```

Comment:
@brianamarie love these bullet point breakdowns, I think there's enough content here maybe we should take them out of bullets and give them each a mini header?

Reply:
Attempted in 52db0afb